resources/design-document.md

```diff
@@ -0,0 +1,267 @@
+
+* Project Creation: An endpoint to create new projects.
+* Project Deletion: An endpoint to delete a project.
+* Project Retrieval: Endpoints for retrieving project details and content.
+* Project Update: Endpoints for updating existing projects.
+* Word Pool Population: Endpoints for fetching content from external APIs to populate a word pool.
+* Workspace Customization: Endpoints for customizing the workspace within a project.
+
+###### Functionality:
+
+* Project Management:
+
+1. Users can create new projects, delete existing projects and retrieve project information.
+2. Projects can be updated as needed to accommodate changes or additions.
+
+* Content Integration:
+
+1. External APIs will be accessed to fetch content, which is then populated into the word pool and workspace.
+
+###### Scalability:
+
+* The serverless architecture offered by AWS Lambda allows for easy scalability. Additional Lambda functions can be added to handle increased load as the project grows.
+
+###### Security:
+
+* Security and authentication will be managed through AWS Cognito to protect user data and project content.
+
+###### Customization:
+
+* Users will have the ability to customize and modify the content within their projects and word pools to suit their specific requirements.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// ProjectModel
+
+String userId;
+String projectId;
+String projectName;
+Date creationDate;
+Date lastModified;
+List<String> wordPool;
+List<String> workspace;
+```
+```
+// WordpoolModel
+
+String userId;
+String wordPoolId;
+String wordPoolName;
+Date creationDate;
+Date lastModified;
+List<String> wordPool;
+
+```
+
+
+## 6.2. Endpoints
+
+### Project
+
+* Accepts POST requests to /project
+###### Accepts a projectName and userId and returns the corresponding ProjectModel including a unique projectId assigned by the Project Service.
+
+* Accepts DELETE requests to /project
+###### Accepts a userId and a projectId and removes a project from user’s projects.
+
+* Accepts GET requests to /project/{projectId}
+###### Accepts a userId and a projectId and returns the corresponding ProjectModel.
+
+* Accepts PUT requests to /project/{projectId}
+###### Accepts data to update a project including an updated projectName.
+
+
+### WordPool
+
+* Accepts POST requests to /wordPool
+###### Accepts a wordPoolName and userId and returns the corresponding WordPoolModel including a unique wordPoolId assigned by the WordPool Service.
+
+* Accepts DELETE requests to /wordPool
+###### Accepts a userId and a wordPoolId and removes a wordPool from user’s wordPools.
+
+* Accepts GET requests to /wordPool/{wordPoolId}
+###### Accepts a userId and a wordPoolId and returns the corresponding WordPoolModel.
+
+* Accepts PUT requests to /wordPool/{wordPoolId}
+###### Accepts data to update a wordPool including an updated wordPoolName.
+
+
```

Comment:
The endpoints above look good for creating/updating individual projects and wordpools. It looks to me like you'll need additional endpoints to support some of the other use cases. For example, how will you associate a wordpool with a project? How will you query back a list of projects?